pkg/openstack/loadbalancer_test.go

```diff
@@ -534,3 +536,104 @@ func Test_getListenerProtocol(t *testing.T) {
 		})
 	}
 }
+
+func TestLbaasV2_getNetworkID(t *testing.T) {
+	lbaas := &LbaasV2{
+		LoadBalancer: LoadBalancer{
+			opts: LoadBalancerOpts{
+				LBClasses: map[string]*LBClass{"lbclassKey": {NetworkID: "lb-class-network-id-1234"}},
+				NetworkID: "lb-class-networkId",
+			},
+		},
+	}
+
+	type fields struct {
+		lbaas LbaasV2
+	}
+	type args struct {
+		service *corev1.Service
+		svcConf *serviceConfig
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		want    string
+		wantErr error
+	}{
+		{
+			name:   "get network id from service annotation",
+			fields: fields{},
+			args: args{
+				service: &corev1.Service{
+					ObjectMeta: v1.ObjectMeta{
+						Annotations: map[string]string{
+							ServiceAnnotationLoadBalancerNetworkID: "subnet-id",
+						},
+					},
+				},
+				svcConf: &serviceConfig{},
+			},
+			want:    "subnet-id",
+			wantErr: nil,
+		},
+		{
+			name: "get network id from config class",
+			fields: fields{
+				lbaas: *lbaas,
+			},
+			args: args{
+				service: &corev1.Service{
+					ObjectMeta: v1.ObjectMeta{
+						Annotations: map[string]string{
+							ServiceAnnotationLoadBalancerClass: "lbclassKey",
+						},
+					},
+				},
+				svcConf: &serviceConfig{},
+			},
+			want:    "lb-class-network-id-1234",
+			wantErr: nil,
+		},
+		{
+			name:   "get network id from default config",
+			fields: fields{},
+			args: args{
+				service: &corev1.Service{},
+				svcConf: &serviceConfig{},
+			},
+			want:    "",
+			wantErr: nil,
+		},
+		{
+			name:   "error when loadbalancer class not found",
+			fields: fields{},
+			args: args{
+				service: &corev1.Service{
+					ObjectMeta: v1.ObjectMeta{
+						Annotations: map[string]string{
+							ServiceAnnotationLoadBalancerClass: "invalid-lb-class",
+						},
+					},
+				},
+				svcConf: &serviceConfig{},
+			},
+			want:    "",
+			wantErr: fmt.Errorf("invalid loadbalancer class \"invalid-lb-class\""),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			lbaas := tt.fields.lbaas
+
+			got, err := lbaas.getNetworkID(tt.args.service, tt.args.svcConf)
+			if (err != nil) != (tt.wantErr != nil) || (err != nil && err.Error() != tt.wantErr.Error()) {
+				t.Errorf("LbaasV2.getNetworkID() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
```

Comment:
Please make sure to use `assert.NoError()` and `assert.ErrorIs()` (or similar function that'd work).